Add trash and restore tests

rpocklin opened the pull request:
Following the merge of https://github.com/owncloud/cdperf/pull/16 here are some tests that leverage the `Trash` and `Restore` functions in K6 load tests.

dragotin reviewed (changes requested):
Thanks for the contribution.

We added a test description doc in `docs/tests.md`.

Do you mind rebasing and adding some description about these new tests? Thanks a lot.

rpocklin commented:
> Thanks for the contribution.
> 
> We added a test description doc in `docs/tests.md`.
> 
> Do you mind rebasing and adding some description about these new tests? Thanks a lot.

Thanks for reviewing @dragotin - I have rebased to `main`, updated the docs and also added large files tests for trash and restore, let me know if you need anything further.

fschade commented:
@rpocklin thanks for the rebase, i checked it out locally and the latest k6 gave me some errors. Can you reproduce those?
![Bildschirmfoto 2022-02-21 um 14 08 18](https://user-images.githubusercontent.com/49308105/154961510-084d0226-cf33-46ee-b5fa-2203e9e9a4ee.png)

rpocklin commented:
> @rpocklin thanks for the rebase, i checked it out locally and the latest k6 gave me some errors. Can you reproduce those? ![Bildschirmfoto 2022-02-21 um 14 08 18](https://user-images.githubusercontent.com/49308105/154961510-084d0226-cf33-46ee-b5fa-2203e9e9a4ee.png)

I did update the dependencies in `package.json` so can you tell me what version of go and k6 you are using?  This PR requires the following import to work:

`import { b64decode } from 'k6/encoding';`

I will re-check with a fresh checkout/build today.

rpocklin commented:
> @rpocklin i used the latest release k6 from their github

@fschade can you tell me the command you tried to run, and also include what is output when you run `k6 version`?

wkloucek commented:
> > @rpocklin i used the latest release k6 from their github
> 
> @fschade can you tell me the command you tried to run, and also include what is output when you run `k6 version`?

@fschade ping

fschade commented:
@wkloucek thanks for the reminder :)

@rpocklin i'm using `k6 v0.36.0 ((devel), go1.17.6, darwin/arm64)` from a pre build binary.
I haven't changed the package.json

steps i did:
```
$ cd src/k6
$ yarn && yarn build
$ k6 run ./dist/test-issue-github-ocis-1018-upload-delete-restore-many-small.js
```

same error out put :( maybe an m1 issue!? I dig a bit deeper now whats goin on here...
![Bildschirmfoto 2022-03-23 um 08 59 23](https://user-images.githubusercontent.com/49308105/159651060-a5f6ade9-cc27-445e-9ce3-b10dbad9ade3.png)

fschade commented:
i upgraded my k6 to the latest version, same here `k6 v0.37.0 ((devel), go1.17.8, darwin/arm64)`

rpocklin commented:
> i upgraded my k6 to the latest version, same here `k6 v0.37.0 ((devel), go1.17.8, darwin/arm64)`

Strange, ran fine for me using your steps (x86_64), additionally I run it as follows (after `yarn && yarn build`):
```
CDPERF_K6_SCRIPTS=$PWD/tests/k6/test-issue-github-ocis-1018-upload-delete-restore-many-small.js

echo $CDPERF_K6_SCRIPTS
./scripts/cdperf --k6-debug=true --cloud-vendor=oc10 --k6-test-host=http://localhost:8080 \
  --k6-docker=false --k6-vus=1 --k6-iterations=1 --k6-duration=1s --k6-scripts=$CDPERF_K6_SCRIPTS
```

dub-rp commented:
I have found the issue, parsing the `fileid` out of the file upload endpoint seems to not be consistent.  I have extended the tests to include a necessary `PROPFIND` call to retrieve this property and use it in both sets of `trash` tests.

Re-tested and works well, for `oc10` and `ocis` versions.

@fschade can you please review again?

rpocklin commented:
I have found the issue, parsing the fileid out of the file upload endpoint seems to not be consistent. I have extended the tests to include a necessary PROPFIND call to retrieve this property and use it in both sets of trash tests.

Re-tested and works well, for oc10 and ocis versions.

@fschade can you please review again?

fschade commented:
@rpocklin thank you, I’m out office till next Monday and have a look then

rpocklin commented:
> @rpocklin thank you, I’m out office till next Monday and have a look then

@fschade any chance of looking at this soon?

rpocklin commented:
Closing due to lack of feedback - happy for Owncloud to pick this up and continue it.
